abci/src/server.rs

```diff
@@ -73,13 +76,21 @@ pub struct Server<App> {
 
 impl<App: Application> Server<App> {
     /// Initiate a blocking listener for incoming connections.
-    pub fn listen(self) -> Result<(), Error> {
-        loop {
-            let (stream, addr) = self.listener.accept().map_err(Error::io)?;
-            let addr = addr.to_string();
-            info!("Incoming connection from: {}", addr);
-            self.spawn_client_handler(stream, addr);
-        }
+    pub fn listen(self) -> Result<(), Error> { 
+
+        let _ = thread::spawn( move || {
+            loop {
+                let (stream, addr) = self.listener.accept().map_err(Error::io).expect( "failed to accept connection");
```

Comment:
Why do we `expect` now instead of returning the error?

Reply:
'Cause of [this line ](https://github.com/rumos-io/tendermint-rs/blob/f7b0a5a4b38dfbb51b51d051907154bc916800a2/abci/src/server.rs#L83C16-L83C119) of code. It blocks the main thread and I chose to move it into a separate thread. 
What do you think:
1. Call expect on error
2. Send error with mpsc
3. Ignore error and log it
4. Don't panic but request cancellation

Reply:
Personally I think. Approach with logging error and stopping application better

Reply:
Agreed